tests/test_job.py

```diff
@@ -308,7 +308,7 @@ def test_interface_job_identity_change(self):
 
     def test_interface_nested_kws(self):
         with pytest.raises(InvalidKeyError):
-            job = self.open_job({"a.b.c": 0})
+            job = self.open_job({"a.b.c": 0}).statepoint
```

Comment:
It's not very user friendly to only warn on statepoint access, but the tradeoff is worth it for the massive speedup

Reply:
Users would be notified on `job.init()` before a numpy array is converted to the statepoint written to disk. 

When using a statepoint to open existing job, the id is calculated the same whether a given item is a numpy array or the equivalent list. In this case, the warning seems unnecessary.